commit 844f7091dc1e7d3fa186017ca67af71340018d29
Author: fang hui <b31070@freescale.com>
Date:   Fri Mar 27 17:38:46 2015 +0800

    MA-6421 - update to lp5.1
    
    Pass compile.
    strncat check more stritly, some array size should add 1.
    
    Signed-off-by: fang hui <b31070@freescale.com>

diff --git a/mx6/libcamera/CameraBridge.cpp b/mx6/libcamera/CameraBridge.cpp
index dc52c5f..cc5f94e 100644
--- a/mx6/libcamera/CameraBridge.cpp
+++ b/mx6/libcamera/CameraBridge.cpp
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2008 The Android Open Source Project
- * Copyright (C) 2012-2014 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2015 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -110,7 +110,7 @@ status_t CameraBridge::getSupportedPictureFormat(int *pFormat,
 
 status_t CameraBridge::initParameters(CameraParameters& params)
 {
-    char tmpBuffer[CAMER_PARAM_BUFFER_SIZE];
+    char tmpBuffer[CAMER_PARAM_BUFFER_SIZE+1];
 
     /*hard code here*/
     params.set(CameraParameters::KEY_FOCUS_DISTANCES, "24.0,50.0,2147483648.0");
diff --git a/mx6/libcamera/OvDevice.h b/mx6/libcamera/OvDevice.h
index 10056e6..aa3413a 100644
--- a/mx6/libcamera/OvDevice.h
+++ b/mx6/libcamera/OvDevice.h
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2008 The Android Open Source Project
- * Copyright (C) 2012 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2015 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -45,8 +45,8 @@ protected:
 
 protected:
     char mSupportedFPS[MAX_SENSOR_FORMAT];
-    char mSupportedPictureSizes[CAMER_PARAM_BUFFER_SIZE];
-    char mSupportedPreviewSizes[CAMER_PARAM_BUFFER_SIZE];
+    char mSupportedPictureSizes[CAMER_PARAM_BUFFER_SIZE+1];
+    char mSupportedPreviewSizes[CAMER_PARAM_BUFFER_SIZE+1];
 };
 
 #endif // ifndef _OV_DEVICE_H_
diff --git a/mx6/libcamera/TVINDevice.h b/mx6/libcamera/TVINDevice.h
index 1dbd3bd..fa6573b 100644
--- a/mx6/libcamera/TVINDevice.h
+++ b/mx6/libcamera/TVINDevice.h
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2008 The Android Open Source Project
- * Copyright (C) 2012-2014 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2015 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -58,8 +58,8 @@ private:
 
 protected:
     char mSupportedFPS[MAX_SENSOR_FORMAT];
-    char mSupportedPictureSizes[CAMER_PARAM_BUFFER_SIZE];
-    char mSupportedPreviewSizes[CAMER_PARAM_BUFFER_SIZE];
+    char mSupportedPictureSizes[CAMER_PARAM_BUFFER_SIZE+1];
+    char mSupportedPreviewSizes[CAMER_PARAM_BUFFER_SIZE+1];
     v4l2_std_id mSTD;
 
     MemmapBuf mMapedBuf[MAX_PREVIEW_BUFFER];
